### ISSUE TITLE
Angular 2 new (RC1) router. Nested routes. Second level router-outlet works only once.

### ISSUE THREAD

**Issue by ctrl-brk** [angular/angular#8731]
Here's the plunker: [http://embed.plnkr.co/B8feYX7e6oxvI2u4zmYW](http://embed.plnkr.co/B8feYX7e6oxvI2u4zmYW/)

Click on "Child 1" - works. Now click on "Child 2" - does not work.
Click "HOME".
Click on "Child 2" - works. Now click on "Child 1" - does not work.

Basically navigation from app root to nested router outlet works only once per app life or until you mandatory navigate back to the root first.

StackOverflow link: [http://stackoverflow.com/questions/37328962/angular-2-new-rc1-router-second-level-router-outlet-works-only-once](http://stackoverflow.com/questions/37328962/angular-2-new-rc1-router-second-level-router-outlet-works-only-once)

Here's some of the code (look at the plunker for the full version):

**app.component**

    @Component({
      selector: 'my-app',
      directives: [ROUTER_DIRECTIVES],
      template: `
        <div>
          <a [routerLink]="['/']">HOME</a>
          <a [routerLink]="['/child', 1]">Child 1</a>
          <a [routerLink]="['/child', 2]">Child 2</a>
        </div>
        <div>
          <router-outlet></router-outlet>
        </div>
      `
    })
    
    @Routes([
        { path: 'child/:id', component: ChildContainerComponent }
    ])    
    
    export class AppComponent { }

**child-container.component**

    @Component({
        directives: [ROUTER_DIRECTIVES],
        template: `
     <div>
        <h3>Child outer container</h3>
        <router-outlet></router-outlet>
    </div>
    `
    })
    
    @Routes([
        { path: '/', component: ChildComponent }
    ])
    
    export class ChildContainerComponent{
    }

**child.component**

    @Component({
        template: `
    <div>
        Child: Id={{_id}}
    </div>
    `
    })

    export class ChildComponent implements OnInit {
        private _id: number;
    
        private get routeSegment(): RouteSegment {  // must be better way to get value
            return this._router.routeTree._root.children[0].value;
        }
    
        constructor(private _router: Router) {
        }
    
        ngOnInit() {
            this._id = +this.routeSegment.getParam('id');
        }
    }

**Comment by sod** [comment]
Works, if you move the `/:id` part of the route to the child container. And you can inject RouteSegment directly.

http://plnkr.co/edit/YI3is1GrYW3Ng6P9N1nk?p=preview

**Comment by ctrl-brk** [comment]
Now, if we go one level deeper - how to do it right?
I added a button within 'child' to navigate to another component and having issues.
Here's the plunker.
[http://embed.plnkr.co/7rUpBQFjlo5VHhofgPbM/](http://embed.plnkr.co/7rUpBQFjlo5VHhofgPbM/)

Just trying to understand this new router.

**Comment by sod** [comment]
`/:id` is kind of catch all. So you have to define more specific routes earlier in the array. And navigate with `router.navigate(['/child/grandchild', 3]);`

http://plnkr.co/edit/AUcSCPBnvuJMJ3D82yV2?p=preview

**Comment by ctrl-brk** [comment]
So, this basically means that every path you pass to router should start from the root?
Can I do relative or something like '..' ?

**Comment by sod** [comment]
```
 * The first segment name can be prepended with `/`, `./`, or `../`.
 * If the segment begins with `/`, the router will look up the route from the root of the app.
 * If the segment begins with `./`, or doesn't begin with a slash, the router will
 * instead look in the current component's children for the route.
 * And if the segment begins with `../`, the router will go up one segment in the url.
```

as of https://github.com/angular/angular/blob/master/modules/@angular/router/src/directives/router_link.ts#L44

**Comment by ctrl-brk** [comment]
Thank you. To make ../.. work I had to pass router segment to the router.navigate as last parameter.
The only question I have now - is there a way to inject previous segment in the constructor like you did with current? I know I can get it in OnActivate but constructor injection requires less code.

**Comment by IgorMinar** [comment]
This issue was made obsolete by router v3: https://angularjs.blogspot.com/2016/06/improvements-coming-for-routing-in.html

If you believe this is not the case, please file a new issue updating the description to router v3. thank you!

**Comment by chenyingyin** [comment]
![default](https://cloud.githubusercontent.com/assets/24284270/20781497/384340e0-b7be-11e6-90d7-cd8e303483ef.png)
![Uploading 图片.png…]()
